kratos/utilities/qef_utility.cpp

```diff
@@ -0,0 +1,115 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Ariadna Cortes
+//
+
+// Project includes
+#include "qef_utility.h"
+
+namespace Kratos {
+
+    array_1d<double,3> QEF::QEFPoint (
+        const GeometryType& rVoxel,  
+        const GeometryArrayType& rTriangles     
+    ) {
+        array_1d<double,3> x_point;
+        array_1d<double,3> center = CalculateCenter(rVoxel);
+        MatrixType m_center(3,1);
+        column(m_center,0) = center;
+        GeometryArrayType edges = rVoxel.GenerateEdges();
+
+        //Initialize the corresponding matrixes
+        MatrixType ata(3,3,0);  //3x3 matrix initialized to 0
+        MatrixType atb(3,1,0);  //3x1 matrix
+        MatrixType btb(1,1,0);  //1x1 matrix
+
+        for (std::size_t i = 0; i < rTriangles.size(); i++) {
+            array_1d<double,3> normal = CalculateNormal(rTriangles[i]);
+            MatrixType m_normal(3,1);
+            column(m_normal,0) = normal;
+
+            //We will iterate through the edges using a while loop, so that if a triangles intersects 2 edges (unlikely 
+            //but possible), only one will be taken into account to create the matrixes.
+            int result = 0; 
+            array_1d<double,3> intersection;
+            std::size_t j = 0;
+            while(!result && j < edges.size()) { 
+                PointsArrayType ends = edges[j++].Points();
+                result = IntersectionUtilities::ComputeTriangleLineIntersection(rTriangles[i],ends[0],ends[1],intersection);
+            }
+            if (result) {
+                //Fill the matrixes with the corresponding information from the intersection and normal
+                MatrixType m_normal_trans = trans(m_normal);
+                ata = ata + prod(m_normal,m_normal_trans);
```

Comment:
```suggestion
                ata += prod(m_normal,m_normal_trans);
```
I'm saying it by heart, but I our matrices support this operator.

Reply:
Yes! thankks